### ISSUE TITLE
Proxy exclusion

### ISSUE THREAD

**Issue by LunaticMuch** [ohmyzsh/ohmyzsh#11571]
### If the feature request is for a plugin or theme, specify it here.

shell-proxy

### If the feature solves a problem you have, specify it here.

When I set my corporate proxy, there's a list of internal domains/URLs, IP where the proxy should not be used. 
If no exclusion is set, the proxy normally returns a `Received HTTP code 502 from proxy after CONNECT` as it's not supposed/configured for handling some internal requests

### Describe the proposed feature.

Shell proxy should also allow the definition of `NO_PROXY` variable, which is the standard in bash shell, and, even better, set it in various softwares (as not always actually correctly read the shell variable for no proxy) 

### Describe alternatives you've considered

So far, I am exporting the proxy exclusion manually and also set it manually in npm and git

**Comment by abhinavms** [comment]
Hi @LunaticMuch,

I wanted to share a quick update on the feature request. I've made some changes related to setting the NO_PROXY variable - https://github.com/ohmyzsh/ohmyzsh/pull/11924.

I've implemented a general change to set the NO_PROXY variable, acknowledging that different software packages handle proxies differently. While this solution may not cover all use cases, I thought it might be a good starting point for improving functionality.

For npm, I currently manage proxy configurations using npmrc.
